src/main/java/org/jabref/logic/exporter/EndnoteXmlExporter.java

```diff
@@ -0,0 +1,218 @@
+import org.jabref.model.entry.types.EntryType;
+import org.jabref.model.entry.types.IEEETranEntryType;
+import org.jabref.model.entry.types.StandardEntryType;
+
+public class EndnoteXmlExporter extends Exporter {
+
+    private static final String ENDNOTE_XML_VERSION = "20.1";
+    private static final String ENDNOTE_NAMESPACE_URI = "http://www.endnote.com/ns/ENdnx";
+    private static final Map<EntryType, String> EXPORT_ITEM_TYPE = Map.ofEntries(
+            Map.entry(StandardEntryType.Article, "Journal Article"),
+            Map.entry(StandardEntryType.Book, "Book"),
+            Map.entry(StandardEntryType.InBook, "Book Section"),
+            Map.entry(StandardEntryType.InCollection, "Book Section"),
+            Map.entry(StandardEntryType.InProceedings, "Conference Paper"),
+            Map.entry(StandardEntryType.Conference, "Conference"),
+            Map.entry(StandardEntryType.MastersThesis, "Thesis"),
+            Map.entry(StandardEntryType.PhdThesis, "Thesis"),
+            Map.entry(StandardEntryType.Proceedings, "Conference Proceedings"),
+            Map.entry(StandardEntryType.TechReport, "Report"),
+            Map.entry(StandardEntryType.Unpublished, "Manuscript"),
+            Map.entry(IEEETranEntryType.Patent, "Patent"),
+            Map.entry(StandardEntryType.Online, "Web Page"),
+            Map.entry(IEEETranEntryType.Electronic, "Electronic Article"),
+            Map.entry(StandardEntryType.Misc, "Generic")
+    );
+
+    private static final Map<EntryType, String> EXPORT_REF_NUMBER = EXPORT_ITEM_TYPE.entrySet().stream()
+                                                                                    .collect(Collectors.toMap(Map.Entry::getKey, entry -> Integer.toString(EXPORT_ITEM_TYPE.entrySet().stream()
+                                                                                                                                                                           .map(Map.Entry::getValue)
+                                                                                                                                                                           .collect(Collectors.toList())
+                                                                                                                                                                           .indexOf(entry.getValue()) + 1)));
+
+    public EndnoteXmlExporter() {
+        super("endnote", "EndNote XML", StandardFileType.XML);
+    }
+
+    @Override
+    public void export(BibDatabaseContext databaseContext, Path file, List<BibEntry> entries) throws Exception {
+        Objects.requireNonNull(databaseContext);
+        Objects.requireNonNull(file);
+        Objects.requireNonNull(entries);
+
+        if (entries.isEmpty()) {
+            return;
+        }
+
+        try (BufferedWriter writer = Files.newBufferedWriter(file, StandardCharsets.UTF_8)) {
+            XMLOutputFactory xof = XMLOutputFactory.newFactory();
```

Comment:
Move the factory initialization to the constructor because it's an expensive operation
         XMLOutputFactory outputFactory = XMLOutputFactory.newFactory();

Reply:
Done.